src/compas_viewer/ui/mainwindow.py

```diff
@@ -19,3 +22,17 @@ def set_window_central(self) -> None:
         central = QWidget()
         central.setLayout(layout)
         self.setCentralWidget(central)
+
+    def set_window_unit(self) -> None:
+        if self.viewer.config.window.unit == "m":
+            grid_size = (10000, 10, 10000, 10)
+        elif self.viewer.config.window.unit == "cm":
+            grid_size = (100, 10, 100, 10)
+        elif self.viewer.config.window.unit == "mm":
+            grid_size = (10, 10, 10, 10)
+        else:
+            print("Invalid unit")
+            return
+
+        self.grid: GridObject = self.viewer.scene.add(Frame.worldXY(), framesize=grid_size, show_frame=True, is_selected=False, is_locked=True, is_visible=True)
+        self.grid.init()
```

Comment:
the grid is still added as an object in the scene. this should not be the case. the grid is part of the view itself...

Reply:
@tomvanmele do you mean sth like `VIewer(show_grid=True)`?